save-orchestrator/src/main/kotlin/com/saveourtool/save/orchestrator/config/KafkaConfiguration.kt

```diff
@@ -0,0 +1,164 @@
+     */
+    @Bean
+    fun producerConfig(): Map<String, Any?> {
+        val producerProps = HashMap(kafkaProperties.buildProducerProperties())
+        producerProps[ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG] = StringSerializer::class.java as Object
+        producerProps[ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG] = JsonSerializer::class.java as Object
+        producerProps[ProducerConfig.PARTITIONER_CLASS_CONFIG] = DefaultPartitioner::class.java as Object
+        return producerProps
+    }
+
+    /**
+     * @return kafka producer factory
+     */
+    @Bean
+    fun producerFactory(): ProducerFactory<Any, Any> = DefaultKafkaProducerFactory(producerConfig())
+
+    /**
+     * @return kafka template
+     */
+    @Bean
+    fun kafkaTemplate(): KafkaTemplate<Any, Any> = KafkaTemplate(producerFactory())
+
+    /**
+     * @return kafka consumer properties
+     */
+    @Bean
+    fun consumerConfig(): Map<String, Any?> {
+        val consumerProps = HashMap(kafkaProperties.buildConsumerProperties())
+        consumerProps[ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG] = "true"
+        consumerProps[ConsumerConfig.AUTO_COMMIT_INTERVAL_MS_CONFIG] = "10"
+        consumerProps[ConsumerConfig.SESSION_TIMEOUT_MS_CONFIG] = "60000"
+        consumerProps[ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG] = ErrorHandlingDeserializer::class.java
+        consumerProps[ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG] = ErrorHandlingDeserializer::class.java
+        consumerProps[ErrorHandlingDeserializer.KEY_DESERIALIZER_CLASS] = StringDeserializer::class.java
+        consumerProps[ErrorHandlingDeserializer.VALUE_DESERIALIZER_CLASS] = JsonDeserializer::class.java
+        consumerProps[ConsumerConfig.PARTITION_ASSIGNMENT_STRATEGY_CONFIG] = listOf(CooperativeStickyAssignor::class.java)
+        consumerProps[JsonDeserializer.TRUSTED_PACKAGES] = "*"
+        return consumerProps
+    }
+
+    /**
+     * @return consumer factory
+     */
+    @Bean
+    fun consumerFactory(): ConsumerFactory<Any, Any> = DefaultKafkaConsumerFactory(consumerConfig())
+
+    /**
+     * @param template
+     * @return kafka listener container factory
+     */
+    @Suppress("GENERIC_VARIABLE_WRONG_DECLARATION")
+    @Bean
+    fun kafkaListenerContainerFactory(template: KafkaTemplate<Any, Any>): ConcurrentKafkaListenerContainerFactory<Any, Any> {
+        val factory = ConcurrentKafkaListenerContainerFactory<Any, Any>()
+        factory.setConsumerFactory(consumerFactory())
+        factory.setConcurrency(1)
+        factory.getContainerProperties().setPollTimeout(LISTENER_POLL_TIMEOUT)
+        factory.setErrorHandler(
+            SeekToCurrentErrorHandler(
+                DeadLetterPublishingRecoverer(template)
+            )
+        )
+        return factory
+    }
+
+    /**
+     * @return kafka agent tasks topic bean
+     */
+    @Bean
+    fun saveAgentTasks(): NewTopic {
+        log.info("Create topic: {} of {} partitions with {} replicas.", requestTopicName)
```

Comment:
it's java style for logging. kotlin way (as I got) is
```suggestion
log.info { "Create topic: $requestTopicName of $missed partitions with $missed replicas." }
```

Reply:
fixed